coral-common/src/main/java/com/linkedin/coral/common/catalog/CoralCatalog.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright 2017-2025 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.common.catalog;
+
+import java.util.List;
+
+
+/**
+ * Top-level catalog interface for accessing datasets in Coral.
+ * This interface provides a unified API for querying table metadata
+ * across different table formats (Hive, Iceberg, etc.).
+ *
+ * CoralCatalog abstracts away the differences between various table formats
+ * and provides a consistent way to access dataset information through
```

Comment:
Ditto on "Dataset" terminology.

Reply:
thanks for the feedback, I have refactored the PR to move away from `Dataset`. 
we now have
(1) `coralCatalog` (new catalog interface) & `HiveMetastoreClient` (old catalog interface) are independent and both work with Coral for translations. `HiveMetastoreClient` has been marked as deprecated in favor of coralCatalog.
(2) `getTable()` is the API in `coralCatalog`. It returns an interface of `CoralTable`. Currently, we have 2 impls of coralTable - `hiveCoralTable` & `icebergCoralTable`